src/julia-syntax.scm

```diff
@@ -512,13 +512,13 @@
           `((|::|
              ;; if there are optional positional args, we need to be able to reference the function name
              ,(if (any kwarg? pargl) (gensy) UNUSED)
-             (call (core kwftype) ,ftype)) (:: ,kw (core NamedTuple)) ,@pargl ,@vararg)
+             (call (core kwftype) ,ftype)) ,kw ,@pargl ,@vararg)
           `(block
             ,(scopenest
               keynames
               (map (lambda (v dflt)
                      (let* ((k     (decl-var v))
-                            (rval0 `(call (top getfield) ,kw (quote ,k)))
+                            (rval0 `(call (top getindex) ,kw (inert ,k)))
```

Comment:
Will constant prop work here? I thought it was only enabled for `getproperty` currently.